### PR TITLE
[Enhancement] check chunk_size value before query execution

### PR DESCRIPTION
**Pull request by silverbullet233**
Why I'm doing:

What I'm doing:

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [x] 3.0
  - [x] 2.5

### CODE REVIEW

File: be/src/runtime/fragment_mgr.cpp [hunk 1]
```diff
@@ -808,6 +808,12 @@ void FragmentMgr::debug(std::stringstream& ss) {
  */
 Status FragmentMgr::exec_external_plan_fragment(const TScanOpenParams& params, const TUniqueId& fragment_instance_id,
                                                 std::vector<TScanColumnDesc>* selected_columns, TUniqueId* query_id) {
+    // check chunk size first
+    auto batch_size = params.batch_size;
+    if (UNLIKELY(batch_size <= 0 || batch_size > MAX_CHUNK_SIZE)) {
+        return Status::InvalidArgument(fmt::format(
+                "batch_size is out of range, it must be in the range (0, 65536], current value is [{}]", batch_size));
```
Comment (kevincai) [inline be/src/runtime/fragment_mgr.cpp:815]:
```suggestion
                "batch_size is out of range, it must be in the range (0, {}], current value is [{}]", MAX_CHUNK_SIZE, batch_size));
```

File: be/src/service/internal_service.cpp [hunk 2]
```diff
@@ -405,6 +406,12 @@ Status PInternalServiceImplBase<T>::_exec_plan_fragment(brpc::Controller* cntl,
         uint32_t len = ser_request.size();
         RETURN_IF_ERROR(deserialize_thrift_msg(buf, &len, request->attachment_protocol(), &t_request));
     }
+    auto batch_size = t_request.query_options.batch_size;
+    if (UNLIKELY(batch_size <= 0 || batch_size > MAX_CHUNK_SIZE)) {
+        return Status::InvalidArgument(fmt::format(
+                "batch_size is out of range, it must be in the range (0, 65536], current value is [{}]", batch_size));
```
Comment (kevincai) [inline be/src/service/internal_service.cpp:412]: ditto

**Review by kevincai — DISMISSED** [review]

**Review by liuyehcf — DISMISSED** [review]

File: be/src/common/constexpr.h [hunk 3]
```diff
@@ -18,6 +18,7 @@ namespace starrocks {
 
 // default value of chunk_size, it's a value decided at compile time
 constexpr const int DEFAULT_CHUNK_SIZE = 4096;
+constexpr const int MAX_CHUNK_SIZE = 65536;
```
Comment (imay) [inline be/src/common/constexpr.h:21]: should be 65535

Comment (kangkaisen) [inline be/src/common/constexpr.h:21]: 65535

**Review by kevincai — DISMISSED** [review]

File: be/src/runtime/fragment_mgr.cpp [hunk 4]
```diff
@@ -808,6 +808,13 @@ void FragmentMgr::debug(std::stringstream& ss) {
  */
 Status FragmentMgr::exec_external_plan_fragment(const TScanOpenParams& params, const TUniqueId& fragment_instance_id,
                                                 std::vector<TScanColumnDesc>* selected_columns, TUniqueId* query_id) {
+    // check chunk size first
+    auto batch_size = params.batch_size;
```
Comment (kangkaisen) [inline be/src/runtime/fragment_mgr.cpp:812]: batch_size is optional，you need check it

File: be/src/service/internal_service.cpp [hunk 5]
```diff
@@ -405,6 +406,13 @@ Status PInternalServiceImplBase<T>::_exec_plan_fragment(brpc::Controller* cntl,
         uint32_t len = ser_request.size();
         RETURN_IF_ERROR(deserialize_thrift_msg(buf, &len, request->attachment_protocol(), &t_request));
     }
+    auto batch_size = t_request.query_options.batch_size;
```
Comment (kangkaisen) [inline be/src/service/internal_service.cpp:409]: batch_size is optional，you need check it

File: be/src/service/internal_service.cpp [hunk 6]
```diff
@@ -405,6 +406,16 @@ Status PInternalServiceImplBase<T>::_exec_plan_fragment(brpc::Controller* cntl,
         uint32_t len = ser_request.size();
         RETURN_IF_ERROR(deserialize_thrift_msg(buf, &len, request->attachment_protocol(), &t_request));
     }
+    if (UNLIKELY(!t_request.query_options.__isset.batch_size)) {
```
[the remark this answers is not in the document]
Comment (silverbullet233) [reply be/src/service/internal_service.cpp:409]: > There will be problems with grayscale upgrade。

why? in what scenarios do we not actively set batch_size?

File: be/src/service/internal_service.cpp [hunk 7]
```diff
@@ -49,6 +49,7 @@
 #include "column/stream_chunk.h"
 #include "common/closure_guard.h"
 #include "common/config.h"
+#include "common/constexpr.h"
```
Comment (trueeyu) [inline be/src/service/internal_service.cpp:52]: why add the header file?

**Review by trueeyu — DISMISSED** [review]

File: be/src/service/internal_service.cpp (hunk 7 above)
Comment (silverbullet233) [reply be/src/service/internal_service.cpp]: > why add the header file?

removed

**Review by kangkaisen — APPROVED** [review]

**Review by liuyehcf — APPROVED** [review]

**Review by kevincai — APPROVED** [review]

**Review by stdpain — APPROVED** [review]

**Pull request merged**